### DISCUSSION TITLE
Upgrade docker image base

### DISCUSSION THREAD

**Discussion by hungran** [pipe-cd/pipecd#4702] [General]
Because why not?
![telegram-cloud-photo-size-5-6271578153075980741-y](https://github.com/pipe-cd/pipecd/assets/26101787/fdd62b9b-67af-4535-9a3b-e9c05e07a8d4)
@khanhtc1202 wdyt?

**Comment by khanhtc1202** [comment]
sure, please 🫡

**Comment by hungran** [comment]
> sure, please 🫡

@khanhtc1202  question?
~isn't nightly build for `Piped` base image?
`tool/piped-base/Dockerfile`~
https://github.com/pipe-cd/pipecd/blob/5ece2e0ecb920a44836bec9e6d3137c3e87a7c77/tool/piped-base/Dockerfile



thought that should be have a flexible way to build image before actual make change on it

**Comment by hungran** [comment]
few thoughts about strategy to build image

- seem pipeCD images has fragment in separate phase under `tool` then `cmd` that might had for new contributor to understand and playaround on local environment --> ideally could try `quickstart` however in view point of development should be even better
- the image base under tool has fix alpine version which turn me think about security risk
Ideally we could combine all of those by multi-stage build, make flexible way to adapt new version for base image when need?
@khanhtc1202
